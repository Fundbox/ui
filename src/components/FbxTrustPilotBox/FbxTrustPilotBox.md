# FbxTrustPilotBox

https://www.trustpilot.com/
display Trust box widget
