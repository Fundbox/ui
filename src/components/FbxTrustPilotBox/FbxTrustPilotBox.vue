<template>
  <div class="trustpilot-widget" :data-locale="locale" :data-template-id="templateId" :data-businessunit-id="businessunitId" :data-style-height="height" :data-style-width="width" data-theme="light">
    <a class="fallback" :href="fallbackLink" target="_blank" rel="noopener">Trustpilot</a>
  </div>
</template>

<script>

import { loadScript } from '../../utils/script-loader.js'

function getTrustPilotUrl(version) {
  return `//widget.trustpilot.com/bootstrap/${version}/tp.widget.bootstrap.min.js`
}

export default {
  name: 'FbxTrustPilotBox',
  props: {
    templateId: {
      type: String,
      required: true,
    },
    businessunitId: {
      type: String,
      required: true,
    },
    fallbackLink: {
      type: String,
      required: true
    },
    height: {
      type: String,
      default: '150px',
    },
    width: {
      type: String,
      default: '100%',
    },
    locale: {
      type: String,
      default: 'en-US',
    },
    version: {
      type: String,
      default: 'v5',
    },

  },
  async mounted() {
    await loadScript(getTrustPilotUrl(this.version))
    window.Trustpilot.loadFromElement(this.$el)
  },
}

</script>
