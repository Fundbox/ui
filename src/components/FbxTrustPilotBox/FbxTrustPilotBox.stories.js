import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxTrustPilotBox from './FbxTrustPilotBox.vue'
import summary from './FbxTrustPilotBox.md'

const stories = storiesOf('Components/TrustPilotBox', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxTrustPilotBox },
  template: `<fbx-trust-pilot-box fallback-link="'https://www.trustpilot.com/review/fundbox.com'" template-id="'53aa8807dec7e10d38f59f32'" businessunit-id="'53b56ea200006400057906ac'" height="'80px'" width="'100px'" style="margin-right: 50px"></fbx-trust-pilot-box>`
})))
