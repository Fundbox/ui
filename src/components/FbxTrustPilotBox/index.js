import FbxTrustPilotBox from './FbxTrustPilotBox.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxTrustPilotBox
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
