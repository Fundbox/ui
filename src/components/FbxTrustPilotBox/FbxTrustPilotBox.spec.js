import { shallowMount } from '@vue/test-utils'
import FbxTrustPilotBox from './FbxTrustPilotBox.vue'

describe('Components/FbxTrustPilotBox', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxTrustPilotBox, {
        // mounting options https://vue-test-utils.vuejs.org/api/options.html
        propsData: {
          templateId: 'ddd',
          businessunitId: 'dds',
          fallbackLink: 'ddsd'
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
