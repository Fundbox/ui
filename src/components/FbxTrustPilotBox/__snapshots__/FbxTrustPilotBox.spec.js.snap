// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxTrustPilotBox snapshots renders the default correctly 1`] = `<div data-locale="en-US" data-template-id="ddd" data-businessunit-id="dds" data-style-height="150px" data-style-width="100%" data-theme="light" class="trustpilot-widget"><a href="ddsd" target="_blank" rel="noopener" class="fallback">Trustpilot</a></div>`;
